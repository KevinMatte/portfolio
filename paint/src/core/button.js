import {useState} from "react";
import React from "react";

import MaterialButton from "@material-ui/core/Button";

import HelpText from './helpText';
import {BaseController} from "./baseController";

export class Controller extends BaseController {
}

export function Button(props) {
    let hooks = {
      helpValue: useState(null),
    };
    let controller = Controller.getController(Controller, props, hooks);

    HelpText.setupController(controller, "helpValue");
    return (
        <div>
            <MaterialButton {...controller.targetProps}>
                {controller.targetProps.children}
            </MaterialButton>
            <HelpText {...controller.helpProps} />
        </div>
    );
}

Button.defaultProps = {
    helpText: null,
};

Button.propTypes = {
    ...HelpText.addedPropTypes,
};

export default Button;
