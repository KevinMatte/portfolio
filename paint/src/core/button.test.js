// Cell.react.test.js
import React from 'react';
import {cleanup, render} from 'react-testing-library';
import 'jest-dom/extend-expect'
import {act} from 'react-dom/test-utils';
import Button, {Controller} from './button';
// import renderer from 'react-test-renderer';

// automatically unmount and cleanup DOM after the test is finished.
// noinspection JSUnresolvedFunction
afterEach(cleanup);

// noinspection ES6ModulesDependencies
test('Button', () => {
    let tree, value;
    let controller = new Controller({});

    const {container, getByTestId, asFragment} = render(
        <Button controller={controller}>Start</Button>,
    );

    tree = asFragment();
    // noinspection JSUnresolvedFunction
    expect(tree).toMatchSnapshot();

    act(() => {
        controller.targetProps.onMouseEnter({type: "mouseenter", currentTarget: {}});
        tree = asFragment();
    });
    expect(tree).toMatchSnapshot();
});

