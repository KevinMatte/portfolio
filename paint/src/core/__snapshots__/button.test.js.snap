// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
      controller="[object Object]"
      name="helpValue"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-2"
      >
        Start
      </span>
      <span
        class="MuiTouchRipple-root-32"
      />
    </button>
    <div />
  </div>
</DocumentFragment>
`;

exports[`Button 2`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
      controller="[object Object]"
      name="helpValue"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-2"
      >
        Start
      </span>
      <span
        class="MuiTouchRipple-root-32"
      />
    </button>
    <div />
  </div>
</DocumentFragment>
`;
